app/src/main/java/org/kiwix/kiwixmobile/webserver/wifi_hotspot/HotspotService.kt

```diff
@@ -0,0 +1,143 @@
+ *
+ */
+package org.kiwix.kiwixmobile.webserver.wifi_hotspot
+
+import android.app.Service
+import android.content.Intent
+import android.os.Binder
+import android.os.IBinder
+import android.widget.Toast
+import org.kiwix.kiwixmobile.KiwixApp
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.registerReceiver
+import org.kiwix.kiwixmobile.core.utils.ServerUtils.getSocketAddress
+import org.kiwix.kiwixmobile.webserver.WebServerHelper
+import org.kiwix.kiwixmobile.webserver.ZimHostCallbacks
+import org.kiwix.kiwixmobile.webserver.ZimHostFragment
+import javax.inject.Inject
+
+/**
+ * HotspotService is used to add a foreground service for the wifi hotspot.
+ * Created by Adeel Zafar on 07/01/2019.
+ */
+class HotspotService :
+  Service(),
+  IpAddressCallbacks,
+  HotspotStateReceiver.Callback {
+  @set:Inject
+  var webServerHelper: WebServerHelper? = null
+
+  @set:Inject
+  var hotspotNotificationManager: HotspotNotificationManager? = null
+
+  @set:Inject
+  var hotspotStateReceiver: HotspotStateReceiver? = null
+
+  private var zimHostCallbacks: ZimHostCallbacks? = null
+  private val serviceBinder: IBinder = HotspotBinder()
+
+  override fun onCreate() {
+    (this.applicationContext as KiwixApp).kiwixComponent
+      .serviceComponent()
+      .service(this)
+      .build()
+      .inject(this)
+    super.onCreate()
+    this.registerReceiver(hotspotStateReceiver!!)
+  }
+
+  override fun onDestroy() {
+    unregisterReceiver(hotspotStateReceiver)
+    super.onDestroy()
+  }
+
+  override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+    when (intent.action) {
+      ACTION_START_SERVER ->
+        intent.getStringArrayListExtra(ZimHostFragment.SELECTED_ZIM_PATHS_KEY)?.let {
+          if (webServerHelper?.startServerHelper(it) == true) {
+            zimHostCallbacks?.onServerStarted(getSocketAddress())
+            startForegroundNotificationHelper()
+            Toast.makeText(
+              this, R.string.server_started_successfully_toast_message,
+              Toast.LENGTH_SHORT
+            ).show()
+          } else {
+            zimHostCallbacks?.onServerFailedToStart()
+          }
+        } ?: kotlin.run {
+          zimHostCallbacks?.onServerFailedToStart()
+        }
+      ACTION_STOP_SERVER -> {
+        Toast.makeText(
+          this, R.string.server_stopped_successfully_toast_message,
+          Toast.LENGTH_SHORT
+        ).show()
+        stopHotspotAndDismissNotification()
+      }
+      ACTION_CHECK_IP_ADDRESS -> webServerHelper?.pollForValidIpAddress()
+      else -> {}
+    }
+    return START_NOT_STICKY
+  }
+
+  override fun onBind(intent: Intent?): IBinder = serviceBinder
+
+  // Dismiss notification and turn off hotspot for devices>=O
+  private fun stopHotspotAndDismissNotification() {
+    webServerHelper?.stopAndroidWebServer()
+    zimHostCallbacks?.onServerStopped()
+    stopForeground(true)
```

Comment:
This method is deprecated in Android 13. We should try to use an alternative.
```
public final void stopForeground (int notificationBehavior)
```

Reply:
Hi @gouri-panda , Done